.ci/onnx/jenkins/runCI.sh

```diff
@@ -16,104 +16,287 @@
 # limitations under the License.
 # ******************************************************************************
 
-NGRAPH_ONNX_REPO="https://github.com/NervanaSystems/ngraph-onnx"
-CI_PATH="$(pwd)"
-CI_ROOT=".ci/onnx/jenkins"
-REPO_ROOT="${CI_PATH%$CI_ROOT}"
-DOCKER_CONTAINER="ngraph-onnx_ci_reproduction"
-
-# Function run() builds image with requirements needed to build ngraph and run onnx tests, runs container and executes tox tests
-function run() {
-    set -x
-    set -e
-
-    cd ./dockerfiles
-    docker build --build-arg http_proxy=$http_proxy --build-arg https_proxy=$https_proxy -f=./ubuntu_16_04.dockerfile -t ngraph-onnx:ubuntu-16_04 .
-
-    cd "${CI_PATH}"
-    if [[ -z $(docker ps -a | grep -i "${DOCKER_CONTAINER}") ]];
-    then
-        docker run -h "$(hostname)" --privileged --name "${DOCKER_CONTAINER}" -v "${REPO_ROOT}":/root \
-            -d ngraph-onnx:ubuntu-16_04 tail -f /dev/null
-        BUILD="TRUE"
+HELP_MESSAGE="Usage: runCI.sh [ARGS] [--help]
+
+Builds nGraph and runs nGraph-ONNX tests.
+
+Arguments:
+    --help                      - Displays this message.
+    --cleanup                   - Removes docker container and files created during script execution, instead of running CI.
+    --rebuild                   - Rebuilds nGraph and runs tox tests.
+    --ngraph-onnx-branch=...    - Will use specified branch of nGraph-ONNX.
+                                  Default: master
+    --ngraph-onnx-sha=...       - Checkout to specified nGraph-ONNX commit.
+                                  Default: none - latest commit of cloned branch used
+    --backends=...              - Comma separated list (no whitespaces!) of nGraph backends to run CI on.
+                                  Default: cpu,interpreter
+"
+
+NGRAPH_ONNX_REPO_ADDRESS="https://github.com/NervanaSystems/ngraph-onnx.git"
+NGRAPH_ONNX_REPO_DIR_NAME="ngraph-onnx"
+NGRAPH_ONNX_CI_DIR=".ci/jenkins"
+
+DEFAULT_NGRAPH_ONNX_REPO_BRANCH="master"
+DEFAULT_NGRAPH_ONNX_REPO_SHA=""
+
+# "<OPERATING_SYSTEM>" will be replaced by particular OS during script execution
+DOCKER_CONTAINER_NAME_PATTERN="ngraph-onnx_ci_reproduction_<OPERATING_SYSTEM>"
+DOCKER_IMAGE_NAME_PATTERN="aibt/aibt/ngraph/<OPERATING_SYSTEM>/base"
+DOCKER_BASE_IMAGE_TAG="ci"
+DOCKER_EXEC_IMAGE_TAG="ci_run"
+DOCKER_HOME="/home/${USER}"
+
+DEFAULT_BACKENDS="cpu interpreter"
+HTTP_PROXY="${http_proxy}"
+HTTPS_PROXY="${https_proxy}"
+
+function main() {
+    # Main function
+
+    # Load parameters defaults
+    CLEANUP="false"
+    REBUILD="false"
+    NGRAPH_ONNX_REPO_BRANCH="${DEFAULT_NGRAPH_ONNX_REPO_BRANCH}"
+    NGRAPH_ONNX_REPO_SHA="${DEFAULT_NGRAPH_ONNX_REPO_SHA}"
+    BACKENDS="${DEFAULT_BACKENDS}"
+
+    parse_arguments "${@}"
+
+    # Calculate necessary paths
+    NGRAPH_CI_ABS_PATH="$(pwd)/$( dirname "${BASH_SOURCE[0]}" )"
+    NGRAPH_ROOT_ABS_PATH="${NGRAPH_CI_ABS_PATH%/.ci*}"
+    local ngraph_parrent_path="$(dirname ${NGRAPH_ROOT_ABS_PATH})"
```

Comment:
`parrent` -> `parent`
```suggestion
    local ngraph_parent_path="$(dirname ${NGRAPH_ROOT_ABS_PATH})"
```

Reply:
fixed